.github/workflows/actions.yml

```diff
@@ -1,22 +1,44 @@
 name: CI Tests
 
-on: [push, pull_request]
+# Pull request will trigger when PR is opened or updated, and more (https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request)
+# Thus limit push checks to master branch so things run after the PR is merged. Without the limit multiple actions run on each PR
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
 
 jobs:
   build:
     runs-on: ubuntu-latest
     timeout-minutes: 30
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - uses: actions/setup-node@v3
         with:
           node-version: 20
       - name: Enable Corepack
         run: corepack enable
       - name: Setup
         run: npm ci
-      - name: Check Code Style
-        run: npm run lint
+      - name: Run eslint on changed files
+        # Always run both eslint and prettier
+        if: always()
```

Comment:
Always run both lint steps, as you might as well know all the issues in one go. In my experience its super annoying to think you fixed all issues and then another tool starts yelling at you.